tests/unit/model/test_graph_objects.py

```diff
@@ -61,3 +64,16 @@ def test_relationship_into_ingest():
 def test_node_key_validity(keys, expected):
     node = Node("Person", keys)
     assert_that(node.has_valid_id, equal_to(expected))
+
+
+def test_get_cached_timestamp():
+    # The first and second timestamp should be the same if called in quick
+    # succession.
+    first = get_cached_timestamp()
```

Comment:
I think the test would be better if we could mock or inject the time values to test different edge cases.

For example, this test works for two timestamps close together and also 2 seconds apart.
But what if I'm not sure what happens when timestamps come down on different sides of the break. Ex: 1.5 and 1.6 or 1.9 and 2.0.

Reply:
Also it could add brittleness. For example, if there is a bug in the code making the test fail only at a certain time, you wouldn't be able to reproduce it, the build would break, but also we probably don't care if 1 out of 1000 timestamps doesn't cache the same as the rest...

Reply:
Yeah... fair point... I'll take another crack at this.